zio/bzngio/reader.go

```diff
@@ -20,25 +20,36 @@ const (
 type Reader struct {
 	peeker *peeker.Reader
 	zctx   *resolver.Context
+	mapper *resolver.Mapper
 }
 
 func NewReader(reader io.Reader, zctx *resolver.Context) *Reader {
 	return &Reader{
 		peeker: peeker.NewReader(reader, ReadSize, MaxSize),
-		zctx:   zctx,
+		zctx:   resolver.NewContext(),
```

Comment:
As I am less familiar with this code, I black box-tested these changes with using multiple bzng files. Unfortunately, I backed this patch out without backing out the ztest you added. I ran the following commands, and the test failed.

```
$ go test -count=1 -short ./...
$ ZTEST_ZQ=$(pwd)/dist/zq go test -v -count=1 -tags=system ./tests -args PATH=$(pwd)/dist
```

This makes me nervous that the test isn't going to catch a regression in this area again.

Reply:
So you backed out all the bzng changes except for my `mulitple` ztest and the ztest changes that allow one to specify multiple inputs in a test?

Reply:
```bash
$ git reset --hard 8e4faf0 && git diff a83ca90..HEAD -- scanner/combiner.go | git apply -R
$ make build
$ go test -count=1 -short ./...
$ ZTEST_ZQ=$(pwd)/dist/zq go test -v -count=1 -tags=system ./tests -args PATH=$(pwd)/dist
```

Reply:
If the procedure for fixing a regression is:
1. Write a failing test
2. Apply a fix to make the test pass

Then I'd expect the above to result in a failed test. They don't; the tests pass. This makes me suspicious that the test is going to catch future regressions. After all, if the test passes when the bug is present, how is it going to fail if the bug is back?

Reply:
Did you back out the changes to bzngio/reader.go as well?

Reply:
Sure, but your change script doesn't back out the changes to `bzngio.Reader`- where the real change happens. The changes to `scanner/combiner.go` only get rid of a `resolver.Mapper` that is now functionally a nop. I ran the inputs in the test against zq master and produced the same bug that was originally reported.

Reply:
🤦‍♂ So I didn't even *notice* this was taking place in `zio/bzingio/reader.go`. I thought all this was in `scanner/combiner.go`.

Reply:
Yep, withdrawn on all this. My apologies. Backing out of *both* those files and rerunning the test causes the test to fail. So sorry for the mistake.